src/js/modules/articles/components/ArticleBody.js

```diff
@@ -138,8 +140,11 @@ const ArticleBody = ({
   );
 };
 
-const mapStateToProps = state => ({
-  articles: state.articles.articles,
-});
 
-export default connect(mapStateToProps)(injectSheet(styles)(ArticleBody));
+export default graphql(ReferencedArticleQuery, {
+  // skip this query if no referencedArticleId was found in article content
+  skip: ({ referencedArticleId }) => referencedArticleId < 0,
```

Comment:
Ehhh. Just let ArticleReference fetch the data and conditionally render ArticleReference. You're overcomplicating this